skorch/tests/test_net.py

```diff
@@ -2493,32 +2493,45 @@ def initialize_module(self, *args, **kwargs):
     @pytest.mark.parametrize("needs_y, train_split, raises", [
         (False, None, ExitStack()),  # ExitStack = does not raise
         (True, None, ExitStack()),
-        (False, lambda x: (x, x), ExitStack()),
-        (True, lambda x, y: (x, x), ExitStack()),
+        (False, "default", ExitStack()),  # Default parameters for NeuralNet
+        (True, "default", ExitStack()),  # Default parameters for NeuralNet
+        (False, lambda x: (x, x), ExitStack()),  # Earlier this was not allowed
+        (True, lambda x, y: (x, x), ExitStack()),  # Works for custom split
         (True, lambda x: (x, x), pytest.raises(TypeError)),  # Raises an error
     ])
-    def test_train_split_with_nans(self, needs_y, train_split, raises):
+    def test_passes_y_to_train_split_when_not_none(
+            self, needs_y, train_split, raises):
         from skorch.net import NeuralNet
         from skorch.toy import MLPModule
 
+        # By default, `train_split=CVSplit(5)` in the `NeuralNet` definition
+        if train_split == "default":
```

Comment:
This looks a bit ugly, but I didn't want to introduce a new function/import to this namespace.